src/qibolab/platforms/abstract.py

```diff
@@ -15,6 +18,68 @@
 )
 
 
+@dataclass
+class Qubit:
+    name: str
+    readout_frequency: int = 0
+    drive_frequency: int = 0
+    sweetspot: float = 0
+    peak_voltage: float = 0
+    pi_pulse_amplitude: float = 0
+    T1: int = 0
+    T2: int = 0
+    state0_voltage: int = 0
+    state1_voltage: int = 0
+    mean_gnd_states: complex = 0 + 0.0j
+    mean_exc_states: complex = 0 + 0.0j
+    resonator_polycoef_flux: List[float] = field(default_factory=list)
+
+
+@dataclass
+class ExecutionResults:
+    """Data strucutre to store output from `execute_pulse_sequence`."""
+
+    i: np.ndarray
+    q: np.ndarray
+    sample: np.ndarray
+
+    def msr(self):
+        """Computes MSR value."""
+        return np.sqrt(self.i**2 + self.q**2)
+
+    def phase(self):
+        """Computes phase value."""
+        return np.angle(self.i + 1j * self.q)
+
+    def to_dict(self, average=True, probability=False):
+        """Serialize output in dict.
+        Args:
+            probability (bool): If `True` returns a dictionary of the form
+                                {'probability: [p1, p2, ...] where p_i are
+                                the probability of every shot.
+            average (bool): If `True` returns a dictionary of the form
+                            {'MSR[V]' : v, 'i[V]' : i, 'q[V]' : q, 'phase[rad]' : phase}.
+                            Where each value is averaged over the number shots. If `False`
+                            all the values for each shot are saved.
+        """
+        if probability:
+            return {"probability": 1 - 2 * self.sample.mean()}
```

Comment:
If `sample` is 0 or 1, this can be negative. Maybe you need just `sample.mean()`? If yes, then this will be the probability to measure 1, so make sure to add it in the docstring. Otherwise if we say "the probability of every shot" it is not clear if it is the probability that the shot is 0 or 1.

Reply:
Yeah I think that it could valuable to add an option to save the probability of the ground state or the probability of the excited state